### ISSUE TITLE
sp_BlitzCache @SortOrder = 'executions' only returning 1 row when duplicate hashes exist

### ISSUE THREAD

**Issue by BrentOzar** [BrentOzarULTD/SQL-Server-First-Responder-Kit#1930]
**Version of the script**
Dev branch, most current code, VersionDate = 20190101 (also verified that it happens in the master branch from the Jan release.)

**What is the current behavior?**
Run sp_BlitzCache @SortOrder = 'executions' and it only returns one row, even when multiple queries are in the plan cache. Ran it with @Debug = 1, verified that the temp tables aren't getting populated correctly. 

Looks like it's when the top query by executions has multiple varieties with the same query hash. The insert into ##bou_BlitzCacheProcs is inserting this:

![screen shot 2019-01-26 at 6 56 56 am](https://user-images.githubusercontent.com/245462/51788904-c98eaf00-2137-11e9-9c33-48e599edaf88.png)

And then later, we're deleting duplicates for the same query hash:

```
WITH d AS (
SELECT  SPID,
        ROW_NUMBER() OVER (PARTITION BY SqlHandle, QueryHash ORDER BY ExecutionCount DESC) AS rn
FROM    ##bou_BlitzCacheProcs
WHERE SPID = @@SPID
)
DELETE d
WHERE d.rn > 1
AND SPID = @@SPID
OPTION (RECOMPILE);
```

Which means we end up with only one row in ##bou_BlitzCacheProcs. That delete was added here, when we were trying to fix the weighting:

https://github.com/BrentOzarULTD/SQL-Server-First-Responder-Kit/commit/af71aabba80d73f021fd219af6d6ee4447531f84

**Which versions of SQL Server and which OS are affected by this issue? Did this work in previous versions of our procedures?**
Checked the last several releases, and it's been the same for all of 'em.

**Comment by erikdarlingdata** [comment]
One quick and dirty fix for this might be to do this. Someone smarter could probably figure out a better way, but I lost my mind trying to figure out what was going wrong with the weighting. 

This would mess up the weighting when you sort by executions, obviously. 

```
WITH d AS (
SELECT  SPID,
        ROW_NUMBER() OVER (PARTITION BY SqlHandle, QueryHash ORDER BY ExecutionCount DESC) AS rn
FROM    ##bou_BlitzCacheProcs
WHERE SPID = @@SPID
AND @SortOrder <> 'executions' --Don't dedupe if sorting by executions
)
DELETE d
WHERE d.rn > 1
AND SPID = @@SPID
OPTION (RECOMPILE);
```

**Comment by MrPshaw** [comment]
It seems the code that was previously deleting the duplicates has been commented out per issue #2026. The deletion code was previously added by #1734 to fix #1352. This may have broken #1352?

`/* 
    This block was used to delete duplicate queries, but has been removed.
    For more info: https://github.com/BrentOzarULTD/SQL-Server-First-Responder-Kit/issues/2026
WITH d AS (
SELECT  SPID,
        ROW_NUMBER() OVER (PARTITION BY SqlHandle, QueryHash ORDER BY #sortable# DESC) AS rn
FROM    ##BlitzCacheProcs
WHERE SPID = @@SPID
)
DELETE d
WHERE d.rn > 1
AND SPID = @@SPID
OPTION (RECOMPILE); 
*/`

To do:
- [ ] Test sp_BlitzCache @SortOrder = 'executions'  to see if it is only returning 1 row when duplicate hashes exist. If returning a row per hash, this issue can be closed.

- [ ] Verify CPU Weight still a percentage, particularly in CPU sort. If weighting broken, #1352 and #1734 need to be reevaluated.

**Comment by BrentOzar** [comment]
Well, it's been two years, and we haven't made progress on this one, so closing it.
